### ISSUE TITLE
Recursive pre-scan fails on danling symlink

### ISSUE THREAD

**Issue by Sbte** [irods/irods#4859]
### iRODS Version, OS and Version
iRODS 4.2.6
CentOS 7

### What did you try to do?
Put a directory containing dangling symlinks with the `--link` argument

### Expected behavior
The dangling symlinks are ignored

### Observed behavior (including steps to reproduce, if applicable)
```
$ iput -r -K --link dir
Running recursive pre-scan... remote addresses: xxx.xxx.xxx.xxx ERROR: USER_INPUT_PATH_ERR: No such file or directory
Path = dir/save.sh

remote addresses: xxx.xxx.xxx.xxx ERROR: USER_INPUT_PATH_ERR: No such file or directory
Path = dir/load.sh

pre-scan complete... errors found.
Aborting data transfer.
```

and `ls -la` gives

```
lrwxrwxrwx  1 user user    21 10 jul  2019 load.sh -> ../../scripts/load.sh
lrwxrwxrwx  1 user user    21 10 jul  2019 save.sh -> ../../scripts/save.sh
```

This `scripts` directory does not exist.

This was supposedly fixed by bb4cb508e822794e3aec8eb43140cea3c829f96c, but this commit is present in 4.2.6, so there still appears to be something wrong.

**Comment by trel** [comment]
Yes, that commit was part of #3072 and was included in the 4.2.3 milestone/release.

The commit includes a test that checks for the dangling symlink case - and it is passing.  

https://github.com/irods/irods/blob/bb4cb508e822794e3aec8eb43140cea3c829f96c/scripts/irods/test/test_iput_options.py#L140-L148

So there is something else different about your scenario.  Your case has a missing directory... Maybe that matters for some reason?

**Comment by alanking** [comment]
I believe this was fixed as part #5359. Test added here:  https://github.com/irods/irods/commit/8254cd75a0923e669ea6bd46315ab7b8500be944
